stdlib/public/core/StringIndexConversions.swift

```diff
@@ -12,7 +12,13 @@
 
 extension String.Index {
   /// Creates an index in the given string that corresponds exactly to the
-  /// specified `UnicodeScalarView` position.
+  /// specified position.
+  ///
+  /// If the index passed as `sourcePosition` represents the start of an
+  /// extended grapheme cluster---the element type of a string---then the
+  /// initializer succeeds. If the index instead represents the position of a
+  /// Unicode scalar within an extended grapheme cluster or the position of an
```

Comment:
I don't think this working needs to have anything to do with Unicode scalars. The semantics are specifically that if the index lies on a grapheme boundary, then it is valid. Non-grapheme boundary indices (whether on scalar boundaries or sub-scalar indices) return nil.